CharlieBackend.AdminPanel/Services/StudentGroupService.cs

```diff
@@ -45,10 +45,12 @@ public async Task<IList<StudentGroupViewModel>> GetAllStudentGroupsAsync(string
             var studentGroupsTask = _apiUtil.GetAsync<IList<StudentGroupDto>>($"{_config.Value.Urls.Api.Https}/api/student_groups", accessToken);
             var studentsTask = _studentService.GetAllStudentsAsync(accessToken);
             var mentorsTask = _mentorService.GetAllMentorsAsync(accessToken);
+            var coursesTask = _courseService.GetAllCoursesAsync(accessToken);
```

Comment:
why dont just await without extra variables ?

Reply:
Because this is an asynchronous request to our API. Here we are running multiple asynс tasks at the same time. If we use await for every API call with many concurrent requests, then in the async methods of our API, the asynс operation is useless, as all requests will be executed synchronously